### DISCUSSION TITLE
Using reflinks when possible

### DISCUSSION THREAD

**Discussion by AndrewKvalheim** [parcel-bundler/parcel#7630] [Ideas] [answered]
I have a project that copies several GB of static assets on each build. What would it take to get Parcel to output [reflinks](https://blog.ram.rachum.com/post/620335081764077568) instead when possible? Node.js offers [`COPYFILE_FICLONE`](https://nodejs.org/api/fs.html#file-copy-constants) but I’m not sure how that relates to Parcel’s architecture.

**Comment by mischnic** [comment, marked as answer]
Yes, that would be great. Currently streams are used pretty heavily (and cause an additional slowdown). So once they are replaced with regular `fs.copyFile` calls, we could simply add that flag to use reflinks where possible.

(There's currently also a "bug" in Node where reflinks aren't supported on macOS: https://github.com/libuv/libuv/pull/2578)

**Reply by AndrewKvalheim** [reply]
Are there plans to use native copying instead of streaming all data through the application?
